dkg/mod.go

```diff
@@ -7,12 +7,20 @@ import (
 
 // DKG defines the primitive to start a DKG protocol
 type DKG interface {
-	Listen(players mino.Players, pubKeys []kyber.Point, threshold int) (Actor, error)
+	// Listen starts the RPC. This function should be called on each node that
+	// wishes to participate in a DKG.
+	Listen() (Actor, error)
 }
 
 // Actor defines the primitives to use a DKG protocol
 type Actor interface {
+	// Setup must be first called by ONE of the actor to use the subsequent
+	// functions. It creates the public distributed key and the private share on
+	// each node. Each node represented by a player must first execute Listen().
+	Setup(players mino.Players, pubKeys []kyber.Point, threshold int) error
+
 	Encrypt(message []byte) (K, C kyber.Point, remainder []byte, err error)
-	Decrypt(K, C kyber.Point) ([]byte, error)
+	Decrypt(players mino.Players, K, C kyber.Point) ([]byte, error)
```

Comment:
Isn't the players included in the start result ?

Reply:
No, and the result is only seen by the leader. Since the nodes receive a `[]Address` in the start message, we could build from that a `mino.Players` so that each node has its players. But that seems a bit hacky.

Reply:
What I mean is that a setup has a very well defined set of players so you shouldn't need to provide that information to decode.

Reply:
yes, but the setup is done once on the leader, so **only the leader** has access to this very well defined set of players. The other nodes have access to a slice of addresses, but we need a mino.Players to do the decrypt. 
If we want every actor to have the mino.Players, we could change `listen()` to take the mino.Players instead of taking it from `Setup()`.

Reply:
Could this be useful: `mino.NewAddresses(addrs...)` ?

Reply:
that's perfect 😃

Reply:
I don't know if its a bug or not, but I was surprised that
```go
	for players.AddressIterator().HasNext() {
		addrs = append(addrs, players.AddressIterator().GetNext())
	}
```
wouldn't work and enter an infinite loop with the players returned by `mino.NewAddresses(addrs...)`. This is due to `AddressIterator()` returning each time a new fresh iterator.

Reply:
Yeah you're not supposed to create a new iterator.